library/core/src/intrinsics.rs

```diff
@@ -2119,18 +2180,20 @@ extern "rust-intrinsic" {
 /// the occasional mistake, and this check should help them figure things out.
 #[allow_internal_unstable(const_eval_select)] // permit this to be called in stably-const fn
 macro_rules! assert_unsafe_precondition {
-    ($e:expr) => {
+    ($([$($tt:tt)*])?($($i:ident:$ty:ty),*$(,)?) => $e:expr) => {
```

Comment:
The documentation should be adjusted to explain the new syntax for this macro.

Reply:
> Using `[...]` for type argument is rather strange, can't this be `<...>`?

Angled brackets are not group tokens and `>` is ambiguous with `$tt:tt`.

Reply:
macro docs updated.